src/hooks/useActiveProject.tsx

```diff
@@ -26,6 +29,31 @@ const ActiveProjectContext = createContext({
   isFetched: false,
 } as ActiveProjectContextProps);
 
+const findProjectAndSubjectById = (
+  projectId?: string | null,
+  projects?: Project[],
+  subjects?: Subject[],
+) => {
+  const selectedProject = projects?.find((p) => p.id === projectId);
+  const selectedSubject = subjects?.find((s) => s.projectId === projectId);
+
+  if (!selectedProject || !selectedSubject) {
+    console.warn('Ignoring attempt to set invalid projectId', projectId);
+
+    // Attempt to fallback to first available project
+    const defaultProject = projects?.[0];
+    const defaultSubject = subjects?.find(
+      (s) => s.projectId === defaultProject?.id,
+    );
+    if (!defaultProject || !defaultSubject) {
+      console.warn('Failed attempt to select default projectId', projectId);
+      return { selectedProject: undefined, selectedSubject: undefined };
+    }
+    return { selectedProject: defaultProject, selectedSubject: defaultSubject };
+  }
```

Comment:
General question about how we want to use `console.warn` in this project.  Are these warnings for us, or for the end consumer/developer who might be building with this SDK in their app?  Obviously these are only shown in development mode.  Also, isn't it a normal, supported case that a user may not have any projects, and thus a console warning might be noise?

The PR looks good though, not wanting to hold it up with my questions.  👍

Reply:
Should add something like `[LifeOmicSDK]` prefixes to logs and errors so that user's can quickly identify where the error is coming from?